modules/product/src/main/java/com/opengamma/strata/product/etd/EtdIdUtils.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.etd;
+
+import java.text.NumberFormat;
+import java.time.YearMonth;
+import java.util.Locale;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.SecurityId;
+import com.opengamma.strata.product.common.ExchangeId;
+import com.opengamma.strata.product.common.PutCall;
+
+/**
+ * A utility for generating ETD identifiers.
+ * <p>
+ * An exchange traded derivative (ETD) is uniquely identified by a set of fields.
+ * In most cases, these fields should be kept separate, as on {@link EtdContractSpec}.
+ * However, it can be useful to create a single identifier from the separate fields.
+ * We do not recommend parsing the combined identifier to retrieve individual fields.
+ */
+public final class EtdIdUtils {
+
+  /**
+   * Scheme used for ETDs.
+   */
+  public static final String ETD_SCHEME = "OG-ETD";
+  /**
+   * Prefix for futures.
+   */
+  private static final String FUT_PREFIX = "F/";
```

Comment:
Since strike can be negative, a dash is not a good option. Any other option? I think the slash reads nicely, but take your point about possible use in a URL